DropwizardHelloWorld/src/main/java/com/example/helloworld/CoolStuffResource.java

```diff
@@ -0,0 +1,45 @@
+package com.example.helloworld;
+
+import com.codahale.metrics.annotation.Timed;
+import com.example.helloworld.service.IntelligentModule;
+import com.example.helloworld.service.RetrieveMessageService;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+import java.util.Optional;
+
+/**
+ * Created by ccardona on 4/12/17.
+ */
+
+@Path("/getCoolStuff")
+@Produces(MediaType.TEXT_PLAIN)
+public class CoolStuffResource {
+
+    private final RetrieveMessageService messageService;
+
+    public CoolStuffResource() {
+        Injector injector = Guice.createInjector(new IntelligentModule());
+        messageService = injector.getInstance(RetrieveMessageService.class);
```

Comment:
😍 Love to see you are using Guice.

No need to change anything here, I just need to give you an advice that, for production services, we don't want to have the `Injector` instance created inside a resource object but as part of the application's initialization. This way we could have dependencies injected into resource objects, making them benefit from the DI advantages (like testability). This is not a straightforward thing to do though. In fact, this would require to integrate Guice through a Dropwizard bundle. An example of such a custom bundle is: https://github.com/HubSpot/dropwizard-guice. Again, this is mostly FYI, no need to change anything here (but you are free to do it if you want 😉).